openmp/runtime/src/kmp_tasking.cpp

```diff
@@ -2512,11 +2512,6 @@ void *__kmp_task_reduction_init(int gtid, int num, T *data) {
   KMP_ASSERT(tg != NULL);
   KMP_ASSERT(data != NULL);
   KMP_ASSERT(num > 0);
-  if (nth == 1) {
```

Comment:
Instead of completely removing the branch, it might be better to check if HHT is enabled. If no, then we can safely return here.

Reply:
I'm not sure how important single threaded reduction costs really are, but sure.